presto-common/src/main/java/com/facebook/presto/common/block/ByteArrayBlock.java

```diff
@@ -35,6 +35,13 @@
 import static io.airlift.slice.SizeOf.sizeOf;
 import static java.lang.String.format;
 
+/**
+ * A block that contains positionCount bytes. Only bytes can be read.
+ * Other block methods like getInt(), getLong(), etc. throw UnsupportedOperationException.
+ *
+ * Despite the name, this block contains individual bytes, not byte arrays.
```

Comment:
yeah, the block itself stores its data as a byte array. it's not used for storing byte arrays.